contracts/staking/contracts/src/immutable/MixinHyperParameters.sol

```diff
@@ -0,0 +1,163 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+import "@0x/contracts-utils/contracts/src/LibRichErrors.sol";
+import "@0x/contracts-utils/contracts/src/Ownable.sol";
+import "../interfaces/IStakingEvents.sol";
+import "../libs/LibStakingRichErrors.sol";
+import "./MixinConstants.sol";
+
+
+contract MixinHyperParameters is
```

Comment:
Is the terminology intended to communicate that these must be set prior to the start of an epoch? If so, it doesn't look like this is enforced so perhaps we could note this more explicitly in the function description for `tune` below.

Reply:
Not really. I don't think changing any of these mid-epoch would break things. I called them hyper-parameters because they're the only tweakable parameters in the system.

Reply:
Ah, gotcha. I might be slightly more in favor of a boring name, like `MixinMutableParameters` or `MixinUpgradableParameters`. But if the meaning is obvious to the other protocologists then I'm cool with it. It does have a darn nice ring to it ツ

Reply:
What about just `MixinParams`, `setParams()`, and `getParams()`?